src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -1132,33 +1132,33 @@ public static int IndexOf<T>(T[] array, T value, int startIndex, int count)
                 if (Unsafe.SizeOf<T>() == sizeof(byte))
                 {
                     int result = SpanHelpers.IndexOf(
-                        ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(Unsafe.As<byte[]>(array)), startIndex),
+                        ref Unsafe.AddByteOffset(ref MemoryMarshal.GetArrayDataReference(Unsafe.As<byte[]>(array)), (uint)startIndex),
                         Unsafe.As<T, byte>(ref value),
-                        count);
+                        (uint)count);
```

Comment:
This is moving more code to aggressive inlined callsite that is not an improvement. I think it was better without before this change.

Reply:
Or  leave this part of the change to separate PR where we can look at the impact.

Reply:
Reverted, can follow up